synapse/cortex.py

```diff
@@ -1401,6 +1403,105 @@ async def getDeprLocks(self):
 
         return retn
 
+    async def reqValidStormGraph(self, gdef):
+        for filt in gdef.get('filters', ()):
+            await self.getStormQuery(filt)
+
+        for pivo in gdef.get('filters', ()):
+            await self.getStormQuery(pivo)
+
+        for form, rule in gdef.get('forms', {}).items():
+            if form != '*' and self.model.form(form) is None:
+                raise s_exc.NoSuchForm(name=form)
+
+            for filt in rule.get('filters', ()):
+                await self.getStormQuery(filt)
+
+            for pivo in rule.get('filters', ()):
+                await self.getStormQuery(pivo)
+
+    async def addStormGraph(self, gdef):
+        gdef['iden'] = s_common.guid()
+
+        s_stormlib_graph.reqValidGdef(gdef)
+
+        await self.reqValidStormGraph(gdef)
+
+        return await self._push('storm:graph:add', gdef)
+
+    @s_nexus.Pusher.onPush('storm:graph:add')
+    async def _addStormGraph(self, gdef):
+        s_stormlib_graph.reqValidGdef(gdef)
+
+        await self.reqValidStormGraph(gdef)
+
+        iden = gdef['iden']
+        if self.graphs.get(iden) is not None:
+            return
+
+        self.graphs.set(iden, gdef)
+        if gdef['scope'] == 'global':
+            self.globalgraphs[iden] = gdef
+        else:
+            self.usergraphs[gdef['creatoriden']][iden] = gdef
+
+        await self.feedBeholder('storm:graph:add', {'gdef': gdef})
+        return copy.deepcopy(gdef)
+
+    async def delStormGraph(self, iden):
+        if await self.getStormGraph(iden) is None:
+            mesg = f'No graph projection with iden {iden} exists!'
+            raise s_exc.NoSuchIden(mesg=mesg)
+
+        return await self._push('storm:graph:del', iden)
+
+    @s_nexus.Pusher.onPush('storm:graph:del')
+    async def _delStormGraph(self, iden):
+        gdef = await self.getStormGraph(iden)
+        if gdef is None:
+            return
+
+        if gdef['scope'] == 'power-up':
```

Comment:
we should no-op (or bail?) if a `scope=power-up` graph makes it into any nexus API i think right?

Reply:
yeah this would just delete the currently loaded graph so it would come back on restart, but I suppose that isn't really a useful thing to be able to do.